test/snowflake_generated_csvs.jl

```diff
@@ -0,0 +1,520 @@
+using ChunkedCSV: ChunkedCSV, TestContext, parse_file, GuessDateTime
+using Parsers: PosLen31
+using Test
+using Dates
+using FixedPointDecimals
+
+# The following test cases were produced with the following Snowflake file format:
+
+# CREATE OR REPLACE FILE FORMAT rai_export_format
+#     TYPE = CSV
+#     FIELD_OPTIONALLY_ENCLOSED_BY = '"'
+#     NULL_IF = ()
+#     TIMESTAMP_FORMAT = 'YYYY-MM-DD HH24:MI:SS.FF3 TZHTZM'
+#     DATE_FORMAT = 'YYYY-MM-DD'
+#     TIME_FORMAT = 'HH24:MI:SS'
+#     BINARY_FORMAT = HEX
+#     ENCODING = UTF8
+# ;
+
+@testset "Snowflake-generated CSVs" begin
+    # CREATE OR REPLACE TABLE binary_test_cases (binary BINARY);
+    # INSERT INTO binary_test_cases VALUES
+    #        (                                ''::BINARY),
+    #        (          '1234567890abcdefABCDEF'::BINARY),
+    #        ('00000000000000000000000000000000'::BINARY),
+    #        ('FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF'::BINARY);
+    @testset "binary" begin
+        testctx = TestContext()
+        parse_file(IOBuffer("""
+            "BINARY"
```

Comment:
i think it'd be better to upload the actual CSVs, like for the `strnig_test_cases.csv`, to avoid us accidentally marking our own homework. Is there any reason not to add the files?